pkg/metrics/clustermesh_test.go

```diff
@@ -0,0 +1,104 @@
+cilium_clustermesh_global_services{source_cluster="local-cluster",source_node_name="worker"} 10
+# HELP cilium_clustermesh_remote_cluster_etcd_has_quorum Whether the remote cluster's 'cilium-etcd' cluster has quorum
+# TYPE cilium_clustermesh_remote_cluster_etcd_has_quorum gauge
+cilium_clustermesh_remote_cluster_etcd_has_quorum{source_cluster="local-cluster",source_node_name="worker",target_cluster="remote-cluster-1"} 1
+cilium_clustermesh_remote_cluster_etcd_has_quorum{source_cluster="local-cluster",source_node_name="worker",target_cluster="remote-cluster-2"} 0
+# HELP cilium_clustermesh_remote_cluster_failures The total number of failures related to the remote cluster
+# TYPE cilium_clustermesh_remote_cluster_failures gauge
+cilium_clustermesh_remote_cluster_failures{source_cluster="local-cluster",source_node_name="worker",target_cluster="remote-cluster-1"} 1
+cilium_clustermesh_remote_cluster_failures{source_cluster="local-cluster",source_node_name="worker",target_cluster="remote-cluster-2"} 2
+# HELP cilium_clustermesh_remote_cluster_last_failure_ts The timestamp of the last failure of the remote cluster
+# TYPE cilium_clustermesh_remote_cluster_last_failure_ts gauge
+cilium_clustermesh_remote_cluster_last_failure_ts{source_cluster="local-cluster",source_node_name="worker",target_cluster="remote-cluster-1"} 0
+cilium_clustermesh_remote_cluster_last_failure_ts{source_cluster="local-cluster",source_node_name="worker",target_cluster="remote-cluster-2"} 1.641133511e+09
+# HELP cilium_clustermesh_remote_cluster_nodes The total number of nodes in the remote cluster
+# TYPE cilium_clustermesh_remote_cluster_nodes gauge
+cilium_clustermesh_remote_cluster_nodes{source_cluster="local-cluster",source_node_name="worker",target_cluster="remote-cluster-1"} 3
+cilium_clustermesh_remote_cluster_nodes{source_cluster="local-cluster",source_node_name="worker",target_cluster="remote-cluster-2"} 3
+# HELP cilium_clustermesh_remote_cluster_readiness_status The readiness status of the remote cluster
+# TYPE cilium_clustermesh_remote_cluster_readiness_status gauge
+cilium_clustermesh_remote_cluster_readiness_status{source_cluster="local-cluster",source_node_name="worker",target_cluster="remote-cluster-1"} 1
+cilium_clustermesh_remote_cluster_readiness_status{source_cluster="local-cluster",source_node_name="worker",target_cluster="remote-cluster-2"} 0
+# HELP cilium_clustermesh_remote_clusters The total number of remote clusters meshed with the local cluster
+# TYPE cilium_clustermesh_remote_clusters gauge
+cilium_clustermesh_remote_clusters{source_cluster="local-cluster",source_node_name="worker"} 2
+`
+
+func (s *StatusCollectorTest) Test_clustermeshCollector_Collect(c *C) {
+	collector := newClustermeshCollectorWithClient(&fakeDaemonClient{
+		response: sampleClustermeshHealthResponse,
+	})
+
+	// perform static checks such as prometheus naming convention, number of labels matching, etc
+	lintProblems, err := testutil.CollectAndLint(collector)
```

Comment:
yup, it's the same implementation. Using promtool lint is ok, but feedback loop is a little bit slow, and the metrics should be available :)

Reply:
Yes, the underlying rule is the same. Using promtool lint is ok, but a little slow in feedback loop, and it also requires actual metric output.

Reply:
Ah neat yeah. In general I have been wondering how we could make the promtool linting of all of the metrics apply automatically, because most people who add new metrics are not aware of it and I didn't necessarily know the best way to recommend how to run the linter before.

We can certainly recommend to always add unit tests for each new metric and use `CollectAndLint()` to verify them, but unfortunately if developers aren't aware of this then they can easily skip the step and end up adding metrics which do not conform to the linter. Broadly one mitigation for this is we just make sure everyone in @cilium/metrics checks for this during review. If we could automatically lint all metrics in the daemon somehow then that would also be cool :thinking: